pkg/joinserver/grpc_deviceregistry.go

```diff
@@ -191,6 +191,17 @@ func (srv jsEndDeviceRegistryServer) Set(ctx context.Context, req *ttnpb.SetEndD
 		return nil, errNoDevEUI.New()
 	}
 
+	if ttnpb.HasAnyField(
+		req.FieldMask.GetPaths(),
+		"claim_authentication_code.value",
+		"claim_authentication_code.valid_from",
+		"claim_authentication_code.value_to",
+	) {
+		log.FromContext(ctx).Warn(
+			`Storage of claim authentication code in the Join Server registry is deprecated. Use the Identity Server registry instead`, //nolint:lll
+		)
```

Comment:
This is not a useful warning for the JS operator.

The warning should be returned to the client via response warnings.

Reply:
`response warnings` as in gRPC response?

Reply:
Check out `warning.Add`.